### PR TITLE
Feat/users management

### PR DESCRIPTION
**Pull request by frascuchon**
This PR normalizes user management and basic authentication. 

- Now, rubrix server enables user basic authentication 
- An default user/pass + api key is automatically configured if no auth setup
- Default api key and pass can be override using env vars
- User management based on basic yaml file
- rubrix client uses the default api key if no api key is provided.

TODO:

- Include user management documentation
- update App references in documentation
- Validate user data model

### CODE REVIEW

**Comment by frascuchon** [comment]
I'll start a basic bullet documentation entry. 

@dvsrepo Where can I include the entry in the documentation?

**Comment by frascuchon** [comment]
A minimal brief doc section is added.

@dvsrepo could you please, take a look?

File: docs/getting_started/user_management.md [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+      ELASTICSEARCH: http://elasticsearch:9200
+      RUBRIX_LOCAL_AUTH_USERS_DB_FILE: /config/.users.yaml
+
+    volumes:
+      # We mount the local file .users.yaml in remote container in path /config/.users.yaml
+      - ${PWD}/.users.yaml:/config/.users.yaml
+      
+    ...
+````
+You can reload the rubrix service for refresh container:
```
Comment (dvsrepo) [inline docs/getting_started/user_management.md]:
```suggestion
You can reload the rubrix service to refresh the container:
```

File: docs/getting_started/user_management.md [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+
+````bash
+docker-compose up -d rubrix
+````
+
+If everything go well, you can use configured users in rubrix for data annotation
```
Comment (dvsrepo) [inline docs/getting_started/user_management.md]:
```suggestion
If everything went well, configured users can now log in and their annotations will be tracked with their usernames.
```

**Review by dvsrepo — CHANGES_REQUESTED** [review]
Looks good to me! only thing left would be to transform it into rst

**Comment by frascuchon** [comment]
@dvsrepo  you can check generated documentation [here](https://docs.rubrix.ml/en/feat-users_management/getting_started/setup%26installation.html#next-steps)

**Comment by dvsrepo** [comment]
Looks good to me but maybe David can do a review and add a short introduction and some more wording for each of the steps, indicating when you want to / need to do each step. Also I'd a reference to where the provided  docker-compose is located

**Comment by dcfidalgo** [comment]
I think we should merge the documentation with the "Advanced installation setup using docker" and simply call it "Advanced installation setup" or something like this. Maybe we can merge this PR as is, and then I provide a new PR with the merge and a revision of the "User management" guide. @frascuchon If you are ok with this proposal, just merge this PR, then I will start working on the follow-up PR.

**Pull request merged**